### PR TITLE
fix(logs): Fixing log messages

### PR DESCRIPTION
**Pull request by oakbani**
## Summary
- This PR modifies log messages to be explicit when it's evaluating an experiment and when it's doing so for a Rollout. 

## Test plan
- Added additional unit test to assert audience logs for rollouts. 
- Existing tests should continue to pass.

### CODE REVIEW

File: lib/optimizely.rb [hunk 1]
```diff
@@ -810,7 +810,7 @@ def get_feature_variable_for_variation(feature_flag_key, feature_enabled, variat
           end
         else
           @logger.log(Logger::DEBUG,
-                      "Feature '#{feature_flag_key}' for variation '#{variation['key']}' is not enabled. Returning the default variable value '#{variable_value}'.")
+                      "Feature '#{feature_flag_key}' is not enabled for user '#{user_id}'. Returning the default variable value '#{variable_value}'.")
```
Comment (aliabbasrizvi) [inline lib/optimizely.rb:813]: The changes are incomplete here. 

On line 809 it should say `Variable value is not defined. Returning the default variable value <variable value> for  variable <variable key>`

On line 916 it should say `User <user ID> was not bucketed into experiment or rollout for feature flag <feature flag key>. Returning the default variable value <variable value>`

File: lib/optimizely/audience.rb [hunk 2]
```diff
@@ -24,23 +24,32 @@ module Optimizely
   module Audience
     module_function
 
-    def user_in_experiment?(config, experiment, attributes, logger)
-      # Determine for given experiment if user satisfies the audiences for the experiment.
+    def user_meets_audience_conditions?(config, experiment, attributes, logger, logging_hash = nil, logging_key = nil)
```
Comment (aliabbasrizvi) [inline lib/optimizely/audience.rb:27]: nit. `does_user_meet_audience_conditions`

File: lib/optimizely/audience.rb [hunk 3]
```diff
@@ -24,23 +24,32 @@ module Optimizely
...
+      # Determine for given experiment/rollout rule if user satisfies the audiences.
```
Comment (aliabbasrizvi) [inline lib/optimizely/audience.rb:28]: nit. `satisfies the audience conditions`.

File: lib/optimizely/audience.rb [hunk 4]
```diff
@@ -24,23 +24,32 @@ module Optimizely
...
       #
       # config - Representation of the Optimizely project config.
-      # experiment - Experiment for which visitor is to be bucketed.
+      # experiment - Experiment/Rollout rule for which visitor is to be bucketed.
```
Comment (aliabbasrizvi) [inline lib/optimizely/audience.rb:31]: nit. `...in which user is to be bucketed`

File: lib/optimizely/audience.rb (hunk 2 above)
Comment (oakbani) [reply lib/optimizely/audience.rb:27]: I can rename it to match with other SDKs but the idiomatic code for ruby is to name methods like these with a question mark. See for example this link with for is_ and has_. https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/PredicateName

**Pull request merged**